Sources/CNIODarwin/shim.c

```diff
@@ -90,4 +91,9 @@ const int CNIODarwin_IPTOS_ECN_CE = IPTOS_ECN_CE;
 const int CNIODarwin_IPV6_RECVPKTINFO = IPV6_RECVPKTINFO;
 const int CNIODarwin_IPV6_PKTINFO = IPV6_PKTINFO;
 
+uint32_t CNIODarwin_get_local_vsock_cid(int socket) {
+    uint32_t cid = 0;
+    int result = ioctl(socket, IOCTL_VM_SOCKETS_GET_LOCAL_CID, &cid);
```

Comment:
This call doesn't check the return value, which is unlikely to be right. Can we pass the `cid` as an out parameter and return a success or failure boolean? Similarly, this needs to loop on `EINTR`.

Reply:
Done in the latest round of commits.

Reply:
Worth noting that, if we really want to do this using a new `System.ioctl`, that's going to be in `NIOPosix` so we'd need to move the API for getting the local Context ID out of `NIOCore` completely.